src/tpm.rs

```diff
@@ -26,50 +28,114 @@ pub const EXIT_SUCCESS: i32 = 0;
 
 static EMPTYMASK: &'static str = "1";
 
-/*
- * tpm data struct for tpmdata.json file IO
- */
-#[derive(Serialize, Deserialize, Debug)]
-struct TpmData {
-    aik_pw: String,
-    ek: String,
-    owner_pw: String,
-    aik_handle: String,
-    aikmod: String,
-    aikpriv: String,
-    aik: String,
-}
-
 /***************************************************************
 ftpm_initialize.py
 Following are function from tpm_initialize.py program      
 *****************************************************************/
 
 /*
-input: content key in tpmdata
-return: deserlized json object
-getting the tpm data struct and convert it to a json value object to access to a particular field inside the tpm data file
-*/
-fn get_tpm_metadata(content_key: String) -> Option<String> {
-    let t_data = read_tpm_data().unwrap();
-    let t_data_json_value = serde_json::to_value(t_data).unwrap();
-    Some(t_data_json_value[content_key].to_string())
+ * Input: content key in tpmdata
+ * Return: Result wrap value string or error code -1
+ *
+ * Getting the tpm data struct and convert it to a json value object to
+ * retrive a particular value by the given key inside the tpm data.
+ */
+fn get_tpm_metadata_content(key: &str) -> Result<Option<String>, i32> {
```

Comment:
This type is kind of gross.  I think you can do it with a `Result<String, i32>` (see below).  However, the i32 doesn't seem to be meaningful; all you ever put there is -1.

Reply:
Changed to Result<String, String>